README.md

```diff
@@ -5,15 +5,57 @@ Copyright (C) 2016-2021 The Open Library Foundation
 This software is distributed under the terms of the Apache License,
 Version 2.0. See the file "[LICENSE](LICENSE)" for more information.
 
-# Introduction
+## Introduction
 
 This module stores permissions and associations between permissions and users.
 It also maintains a hierarchy of permissions and sub-permissions, allowing for
 permissions to act as roles, rather than simple bits. It is used primarily by
 the Authtoken module, though it is possible that some Authentication
 implementations may have reason to make calls to the Permissions module as well.
 
-# Additional information
+There are two types of permissions:
+
+*  Immutable permissions that are defined in module descriptors. When modules
+   are enabled for a tenant, they are passed from Okapi to mod-permissions via
+   the [tenantPermissions](ramls/tenantPermissions.raml) interface.
+
+*  Mutable, User-defined, permissons that any module can manage if they
+   have permissions to do so. This is part of the
+   [permissions](ramls/permissions.raml) interface.
+
+Mod-permission also has the notion of users and permissions associated
+with them. There are two identifiers for each user - the external identifier
+of users in the users interface and the internal identifier of the user
+in mod-permissions. Having the two is confusing, they could be the same.
```

Comment:
My proposal how to make the "There are two ..." sentence more understandable:

The /perms/users API has two user identifiers: id and userId. id is the mod-permissions identifier for the user while userId points to the id field of the mod-users /users API.